subsys/net/lib/tls_credentials/Kconfig

```
@@ -36,6 +37,22 @@ config TLS_CREDENTIALS_BACKEND_PROTECTED_STORAGE
 
 endchoice
 
+config TLS_CREDENTIAL_CSR
+	bool "Enable CSR generation"
+	select EXPERIMENTAL
+	depends on MBEDTLS
+	help
+	  Allow the TLS Credentials Module to generate a CSR from a private key.
+	  Optionally generate a new key from scratch if TLS_CREDENTIAL_KEYGEN enabled.
+
+config TLS_CREDENTIAL_KEYGEN
+	bool "Enable private/public keypair generation"
+	select EXPERIMENTAL
+	depends on MBEDTLS
+	depends on MBEDTLS_USE_PSA_CRYPTO || !TLS_CREDENTIALS_BACKEND_PROTECTED_STORAGE
```

I think
```suggestion
	depends on MBEDTLS_PSA_CRYPTO_CLIENT || !TLS_CREDENTIALS_BACKEND_PROTECTED_STORAGE
```
is more correct because `CONFIG_MBEDTLS_USE_PSA_CRYPTO` is automatically set whenever there is some PSA crypto provider in the system (this can either be implemented by Mbed TLS itself or TF-M). I think what you need to ask for there is the presence of some PSA Crypto API provider.

---

Ahh, good point. Thank you, I will switch to that.